src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3443,7 +3443,24 @@ private void GetContainerPropertiesInternal()
                     //
                     if (RuntimeId == Guid.Empty)
                     {
-                        ContainerObRoot = (string)computeSystemPropertiesType.GetProperty("ObRoot").GetValue(computeSystemPropertiesHandle);
+                        var obRootfieldInfo = computeSystemPropertiesType.GetField("ObRoot");
+                        if (obRootfieldInfo != null)
+                        {
+                            ContainerObRoot = obRootfieldInfo.GetValue(computeSystemPropertiesHandle) as string;
+                        }
+                        else
+                        {
+                            var obRootPropertyInfo = computeSystemPropertiesType.GetProperty("ObRoot");
+                            if (obRootPropertyInfo != null)
+                            {
+                                ContainerObRoot = obRootPropertyInfo.GetValue(computeSystemPropertiesHandle) as string;
+                            }
+                        }
+
+                        if (ContainerObRoot == null) 
+                        {
+                            throw new ArgumentNullException(nameof(ContainerObRoot));
```

Comment:
I think we should throw the `RemotingErrorIdStrings.CannotGetHostInteropTypes` exception here, same as above.